dashboard/src/modules/components/ProfileComponent/ClipboardCopy.jsx

```diff
@@ -0,0 +1,54 @@
+import { Button, Tooltip } from "@patternfly/react-core";
+import React, { useState } from "react";
+
+import { CopyIcon } from "@patternfly/react-icons";
+import { SHOW_COPIED_TEXT_MS } from "assets/constants/copyTextConstants";
+
+const ClipboardCopy = ({ copyText }) => {
+  const [isCopied, setIsCopied] = useState(false);
+
+  const copyTextToClipboard = async (text) => {
+    if ("clipboard" in navigator) {
+      return await navigator.clipboard.writeText(text);
+    } else {
+      return document.execCommand("copy", true, text);
+    }
```

Comment:
The documentation says that `execCommand` is deprecated, and points at the "modern" `navigator.clipboard`, which shows support across all the modern browsers. Is this fallback actually worthwhile? (I.e., what are the chances we'll run somewhere that doesn't support `navigator.clipboard` but does support `document.execCommand("copy", ...)`?)

With that said, I'm not going to object if you think that's worthwhile!

Reply:
The use of navigator.clipboard requires a secure origin. As our staging environment is being served over HTTP, then the clipboard method won't be available.

I'm aware that `document.execCommand()` is deprecated. But to have copy work in HTTP, I'm using this method.

Reply:
Ah, thanks! I'd like to see that explanation in the code as a comment, because that's not immediately obvious to a reader.

We intend to be switching the front end Pbench Server connection to `https` soon, which should help here if that's the only issue.